File: pdu/python/geometry_msgs/pdu_cdr_conv_Accel.py
from .pdu_pytype_Accel import Accel
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..geometry_msgs.pdu_cdr_conv_Vector3 import *



def py_to_cdr_body_Accel(writer: CdrWriter, src: Accel):
    py_to_cdr_body_Vector3(writer, src.linear)
    py_to_cdr_body_Vector3(writer, src.angular)


def cdr_body_to_py_Accel(reader: CdrReader, dst: Accel):
    cdr_body_to_py_Vector3(reader, dst.linear)
    cdr_body_to_py_Vector3(reader, dst.angular)
    return dst


def py_to_cdr_Accel(src: Accel) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Accel(writer, src)
    return writer.bytes()


def cdr_to_py_Accel(cdr_payload) -> Accel:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Accel()
    return cdr_body_to_py_Accel(reader, dst)

File: pdu/javascript/geometry_msgs/pdu_cdr_conv_Accel.js
import { Accel } from './pdu_jstype_Accel.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Vector3 } from '../geometry_msgs/pdu_jstype_Vector3.js';
import { PduVector3Converter } from '../geometry_msgs/pdu_cdr_conv_Vector3.js';


export class PduAccelConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Accel } src
     */
    static to_cdr_body(writer, src) {
        PduVector3Converter.to_cdr_body(writer, src.linear);
        PduVector3Converter.to_cdr_body(writer, src.angular);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Accel } dst
     * @returns { Accel }
     */
    static cdr_body_to_js(reader, dst) {
        PduVector3Converter.cdr_body_to_js(reader, dst.linear);
        PduVector3Converter.cdr_body_to_js(reader, dst.angular);
        return dst;
    }

    /**
     * @param { Accel } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Accel }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Accel());
    }
}
